src/gpa.py

```diff
@@ -152,32 +156,50 @@ def compute_results(student_records):
    sessions = get_sessions(student_records)
    semesters = get_semesters(student_records)
    session_total_units, session_total_grade_points =[],[]
-   final_result = []
-   final_result.append(f"Student Name: {student_records['name']}")
-   final_result.append(f"Student ID: {student_records['id']}")
 
+
+   results = list()
+   final_result = {}
+   final_result["Student Name"] = student_records['name']
+   final_result["Student ID"] =  student_records["id"]
+
+   gpa_results ={}
+   gpa_results["Student ID"] =  student_records["id"]
```

Comment:
I understand what you're trying to do here but it is not conventional to use free-form strings for dictionary keys. For example, `final_result["Student Name"]` should be more like `final_result["student_name"]` or simply `final_result["name"]`